src/common/routing/hooks/useBackListener.js

```diff
@@ -0,0 +1,20 @@
+import { useContext, useEffect } from "react";
+import { UNSAFE_NavigationContext } from "react-router-dom";
+
+const useBackListener = (callback) => {
+  const { navigator } = useContext(UNSAFE_NavigationContext);
```

Comment:
It will be good to post a comment about `UNSAFE_NavigationContext` like what it is for.

Reply:
I don't know much about it. It would be great if you could add to this

Reply:
We have discussed this on a call today. So hope, we are on the same page on this.